### PR TITLE
fix(hotspot-packages): move to the classic rest/hotspotpackage surface

### PR DESCRIPTION
**Pull request by jpennin5**
## Description

Part of #108 (item B3), which found
`/integration/v1/sites/{site}/hotspot/packages` 404s and could not locate
any Integration v1 route for packages.

**Fix:** the classic surface answers — verified live against Network
10.5.67, `GET /api/s/{site}/rest/hotspotpackage` returns 200. All five
package tools now use it.

**Write-path mapping (from the same controller):** the validator echoes
exactly `{amount, hours, trial_duration_minutes}` on rejection, so those
are the fields it reads: duration is hours-granular (sent as `hours`,
rounded up from `duration_minutes`), price is `amount` with `currency`
alongside. Two validator quirks worth recording for future maintainers:

- with `amount` left at 0 the controller reports
  `api.err.InvalidHotspotPackageDuration` regardless of the actual
  duration, and
- a site with no hotspot payment gateway configured refuses creation
  outright with `api.err.Invalid`.

The previous bandwidth/quota/enabled parameters are removed rather than
remapped: the validator never acknowledged any of them, and keeping
parameters the controller ignores misrepresents the tool. Responses are
returned as raw dicts; the strict `HotspotPackage` model described a
shape no endpoint returns and rejected everything.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

Parameters are removed from tool signatures, but the old route never
worked, so nothing that worked can break.

## Testing

- Tests cover the classic routes, the hours rounding, price/currency
  mapping, and raw-dict returns.
- `pytest tests/unit/` passes; pre-commit hooks pass for all files this
  PR touches.
- `API.md` use-case prose updated (packages are sold by duration; quotas
  are not part of this API surface).

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (API.md, CHANGELOG.md)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published (none — stands alone)

## AI Assistance

This PR was created with assistance from Claude Code.

**Human Review Status:** ✅ Reviewed and approved by @jpennin5
**Testing:** Full unit suite passing; read path and validator behavior
verified against a live Network 10.5.67 controller (write path gated by a
payment gateway, documented above).
**Security Review:** Completed — all fixture values are synthetic.

### CODE REVIEW

File: src/tools/radius.py [hunk 1]
```diff
@@ -852,24 +854,17 @@ async def create_hotspot_package(
...
+        # Build request payload from the fields the classic validator is
+        # known to read (it echoes amount, hours and trial_duration_minutes
+        # on rejection). Duration is hours-granular on this surface.
         payload: dict[str, Any] = {
             "name": name,
-            "duration_minutes": duration_minutes,
-            "currency": currency,
-            "enabled": True,
+            "hours": max(1, -(-duration_minutes // 60)),
         }
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/radius.py:874]: Fixed in 2cde828: `duration_minutes < 1` raises ValidationError on create and update instead of rounding up to a 1-hour paid package.

File: src/tools/radius.py [hunk 2]
```diff
@@ -805,39 +805,41 @@ async def list_hotspot_packages(
         if not client.is_authenticated:
             await client.authenticate()
 
-        response = await client.get(f"/integration/v1/sites/{site_id}/hotspot/packages")
+        response = await client.get(f"/ea/sites/{site_id}/rest/hotspotpackage")
         data = response if isinstance(response, list) else response.get("data", [])
 
-        return [HotspotPackage(**package).model_dump() for package in data]
+        return [dict(package) for package in data]
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/radius.py:817]: You're right — the raw-dict passthrough made those fixtures pass vacuously. Fixed in 2cde828: list/get fixtures now use the classic hours/amount shape and assert the `rest/hotspotpackage` endpoint.

File: src/tools/radius.py [hunk 3]
```diff
@@ -979,21 +963,11 @@ async def update_hotspot_package(
...
-    if download_quota_mb is not None:
-        payload["download_quota_mb"] = download_quota_mb
-    if upload_quota_mb is not None:
-        payload["upload_quota_mb"] = upload_quota_mb
+        payload["hours"] = max(1, -(-duration_minutes // 60))
     if price is not None:
-        payload["price"] = price
+        payload["amount"] = price
     if currency is not None:
         payload["currency"] = currency
```
[the remark this answers is not in the document]
Comment (jpennin5) [reply src/tools/radius.py:988]: Fixed in 2cde828: `currency` without `price` now raises instead of sending a request the classic validator reads as changing nothing.

**Comment by enuno** [comment]
### Summary
Request changes. The PR is not yet merge-ready because the current status checks are not clean.

### Critical findings
- Merge state: \ / \.
- Failing check(s): Mermaid Diagram Sync Assistant.
- The changed tooling surface should be revalidated after the failing check is resolved.

### Suggestions
1. Resolve the failing status check(s) and rerun CI.
2. Reconfirm the changed tooling behavior against the intended controller/fixture shapes.
3. Ensure regression tests cover the original failure mode and edge cases introduced here.
4. Update docs if any request/response or error semantics changed.

### Verification checklist for the author
- [ ] Required checks are passing
- [ ] Regression tests cover the changed path
- [ ] Lint / format / type checks pass locally
- [ ] Documentation is updated where user-facing behavior changed
- [ ] No secrets, credentials, or controller-identifying data were introduced

**Comment by jpennin5** [comment]
The failing check is the `Mermaid Diagram Sync Assistant` GitHub App, not a workflow in this repository. Its check-run output on this PR reads: "Webhook Processing Failed — An unexpected error occurred while processing the webhook. Please try pushing a new commit or check your repository configuration." That processing happens on the app's side; pushing new commits to these PRs has re-triggered the check with the same result, so it is not reachable from a fork.

This is the loop documented in #112: the app has errored on every fork PR since late June, fork CI runs sit at `action_required` awaiting maintainer approval, and the daily automated review then reads the app error as a failing required check and requests changes.

Every check that can conclude on a fork PR is green here (both Socket Security checks). The remaining steps are repo-side — approving the CI runs and either repairing/removing the app integration or unrequiring its check. #117 merged with the identical check state, so the merge path exists. Happy to rebase or adjust anything code-side the moment CI can actually run.
